app/services/movie_service.rb

```diff
@@ -0,0 +1,25 @@
+class MovieService
+  def top_rated
+    movies = []
+    i = 0
+    while movies.count < 40
+      i +=1
+      movies << to_json("movie/top_rated?page=#{i}")[:results]
+      movies.flatten!
+    end
+    movies
+  end
```

Comment:
@iandouglas @dionew1

I think we may learn this in class today but do we ever need to "protect" our code from possible future changes in the API? For example, we could have written the `top_rated` method by simply calling on 2 pages of API data but that assumes the pages of that API will always return 20 items. We added the while loop as a way to protect against that.

Reply:
It's good to consider these edge cases when writing your code and how to handle if the API response does change how it sends results. Something else to consider is if for some reason the top_rated doesn't have 40 results. How can we account for this and have our code handle it gracefully?